pkgs/development/python-modules/wgpu-py/default.nix

```diff
@@ -96,66 +98,62 @@ buildPythonPackage rec {
+  # as there is no "script" to wrap.
+  #
+  # In addition thereto, the structure of the tests in `wgpu-py` is unconventional, spread in separate folders
+  # all in the repository's root, which causes `pytestCheckHook` to fail for Darwin platforms too.
+  # As such, we delegate testing to `passthru`.
   doCheck = false;
 
-  passthru = {
-    tests =
+  passthru.tests =
+    {
+      version = testers.testVersion {
+        package = wgpu-py;
+        command = "python3 -c 'import wgpu; print(wgpu.__version__)'";
+      };
+    }
+    // lib.optionalAttrs stdenv.buildPlatform.isDarwin (
+      let
+        runWGpuPyTestCommand =
+          name: text:
+          pkgs.runCommandNoCC name {
+            nativeBuildInputs = [ wgpu-py ] ++ nativeCheckInputs;
+            env.WGPU_LIB_PATH = "${wgpu-native}/lib/libwgpu_native${stdenv.hostPlatform.extensions.library}";
+          } text;
+      in
       {
-        version = testers.testVersion {
-          package = wgpu-py;
-          command = "python3 -c 'import wgpu; print(wgpu.__version__)'";
-        };
+        tests = runWGpuPyTestCommand "tests" ''
+          pytest -v ${wgpu-py.src}/tests
+        '';
+
+        examples = runWGpuPyTestCommand "examples" ''
+          pytest -v ${wgpu-py.src}/examples
+        '';
+
+        codegen = runWGpuPyTestCommand "codegen" ''
+          pytest -v ${wgpu-py.src}/codegen
+        '';
+
+        tests_mem = runWGpuPyTestCommand "tests_mem" ''
+          pytest -v ${wgpu-py.src}/tests_mem
+        '';
       }
-      // lib.optionalAttrs stdenv.buildPlatform.isDarwin {
-        tests = testers.runCommand {
-          name = "tests";
-          script = ''
-            WGPU_LIB_PATH=${wgpu-native}/lib/libwgpu_native${stdenv.hostPlatform.extensions.library} \
-              pytest -v ${wgpu-py.src}/tests
-          '';
-          nativeBuildInputs = [ wgpu-py ] ++ nativeCheckInputs;
-        };
-
-        examples = testers.runCommand {
-          name = "examples";
-          script = ''
-            WGPU_LIB_PATH=${wgpu-native}/lib/libwgpu_native${stdenv.hostPlatform.extensions.library} \
-              pytest -v ${wgpu-py.src}/examples
-          '';
-          nativeBuildInputs = [ wgpu-py ] ++ nativeCheckInputs;
-        };
-
-        codegen = testers.runCommand {
-          name = "codegen";
-          script = ''
-            WGPU_LIB_PATH=${wgpu-native}/lib/libwgpu_native${stdenv.hostPlatform.extensions.library} \
-              pytest -v ${wgpu-py.src}/codegen
-          '';
-          nativeBuildInputs = [ wgpu-py ] ++ nativeCheckInputs;
-        };
-
-        tests_mem = testers.runCommand {
-          name = "tests_mem";
-          script = ''
-            WGPU_LIB_PATH=${wgpu-native}/lib/libwgpu_native${stdenv.hostPlatform.extensions.library} \
-              pytest -v ${wgpu-py.src}/tests_mem
-          '';
-          nativeBuildInputs = [ wgpu-py ] ++ nativeCheckInputs;
-        };
-      };
-  };
+    );
```

Comment:
Wouldn't an override of `checkPhase` be more appropriate here?

```nix
checkPhase = ''
  runHook preCheck

  export WGPU_LIB_PATH="${wgpu-native}/lib/libwgpu_native${stdenv.hostPlatform.extensions.library}"
  pytest -v tests
  pytest -v examples
  pytest -v codegen
  pytest -v tests_mem

  runHook postCheck
'';
```

Reply:
Maybe [`get_wgpu_lib_path()`](https://github.com/pygfx/wgpu-py/blob/6b77f99b854bf11f97913bbf296e5a584dac1e3e/wgpu/backends/wgpu_native/_ffi.py#L81) should even be patched to always point to `"${wgpu-native}/lib/libwgpu_native${stdenv.hostPlatform.extensions.library}"`

Reply:
Doing this will remove any possibility of overriding the installed wgpu-native library though.
Is it still worth it?